### ISSUE TITLE
SVG problems in deploy

### ISSUE THREAD

**Issue by patrickelectric** [probonopd/linuxdeployqt#284]
I'm almost certainly that everything is correct with all svg dependences, but maybe something is missing.

Building the project from the source code:
![image](https://user-images.githubusercontent.com/1215497/39055162-de39a2b0-4489-11e8-8a76-cdbfe3122f32.png)

After linuxdeployqt:
![image](https://user-images.githubusercontent.com/1215497/39055209-f66f0bb8-4489-11e8-8616-985659cd5f4b.png)

All 3 icons are svg, the one in the middle is completely weird, and the boths in the corner shows with pixels.

export QT_DEBUG_PLUGINS=1;
export QT_LOGGING_RULES="qt.network*.warning=true";
```
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/platforms" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/platforms/libqxcb.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/platforms/libqxcb.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "xcb"
        ]
    },
    "className": "QXcbIntegrationPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("xcb")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/platforms" ...
loaded library "/tmp/.mount_pingviUbFT8p/plugins/platforms/libqxcb.so"
loaded library "Xcursor"
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/xcbglintegrations" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/xcbglintegrations/libqxcb-glx-integration.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/xcbglintegrations/libqxcb-glx-integration.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.Xcb.QXcbGlIntegrationFactoryInterface.5.5",
    "MetaData": {
        "Keys": [
            "xcb_glx"
        ]
    },
    "className": "QXcbGlxIntegrationPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("xcb_glx")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/xcbglintegrations" ...
loaded library "/tmp/.mount_pingviUbFT8p/plugins/xcbglintegrations/libqxcb-glx-integration.so"
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/platformthemes" ...
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/platformthemes" ...
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/platforminputcontexts" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QPlatformInputContextFactoryInterface.5.1",
    "MetaData": {
        "Keys": [
            "compose",
            "xim"
        ]
    },
    "className": "QComposePlatformInputContextPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("compose", "xim")
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QPlatformInputContextFactoryInterface.5.1",
    "MetaData": {
        "Keys": [
            "ibus"
        ]
    },
    "className": "QIbusPlatformInputContextPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("ibus")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/platforminputcontexts" ...
loaded library "/tmp/.mount_pingviUbFT8p/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.so"
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/styles" ...
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/styles" ...
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick.2/libqtquick2plugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuick2Plugin",
    "debug": false,
    "uri": [
        "QtQuick.2"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick.2/libqtquick2plugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick/Controls.2/libqtquickcontrols2plugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuickControls2Plugin",
    "debug": false,
    "uri": [
        "QtQuick.Controls.2"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick/Controls.2/libqtquickcontrols2plugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick/Window.2/libwindowplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuick2WindowPlugin",
    "debug": false,
    "uri": [
        "QtQuick.Window.2"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick/Window.2/libwindowplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick/Templates.2/libqtquicktemplates2plugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuickTemplates2Plugin",
    "debug": false,
    "uri": [
        "QtQuick.Templates.2"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick/Templates.2/libqtquicktemplates2plugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick/Controls.2/Material/libqtquickcontrols2materialstyleplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuickControls2MaterialStylePlugin",
    "debug": false,
    "uri": [
        "QtQuick.Controls.2.Material"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick/Controls.2/Material/libqtquickcontrols2materialstyleplugin.so"
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/imageformats" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/imageformats/libqgif.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/imageformats/libqgif.so, metadata=
{
    "IID": "org.qt-project.Qt.QImageIOHandlerFactoryInterface",
    "MetaData": {
        "Keys": [
            "gif"
        ],
        "MimeTypes": [
            "image/gif"
        ]
    },
    "className": "QGifPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("gif")
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/imageformats/libqico.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/imageformats/libqico.so, metadata=
{
    "IID": "org.qt-project.Qt.QImageIOHandlerFactoryInterface",
    "MetaData": {
        "Keys": [
            "ico",
            "cur"
        ],
        "MimeTypes": [
            "image/vnd.microsoft.icon"
        ]
    },
    "className": "QICOPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("ico", "cur")
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/imageformats/libqjpeg.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/imageformats/libqjpeg.so, metadata=
{
    "IID": "org.qt-project.Qt.QImageIOHandlerFactoryInterface",
    "MetaData": {
        "Keys": [
            "jpg",
            "jpeg"
        ],
        "MimeTypes": [
            "image/jpeg",
            "image/jpeg"
        ]
    },
    "className": "QJpegPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("jpg", "jpeg")
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/imageformats/libqsvg.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/imageformats/libqsvg.so, metadata=
{
    "IID": "org.qt-project.Qt.QImageIOHandlerFactoryInterface",
    "MetaData": {
        "Keys": [
            "svg",
            "svgz"
        ],
        "MimeTypes": [
            "image/svg+xml",
            "image/svg+xml-compressed"
        ]
    },
    "className": "QSvgPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("svg", "svgz")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/imageformats" ...
loaded library "/tmp/.mount_pingviUbFT8p/plugins/imageformats/libqgif.so"
loaded library "/tmp/.mount_pingviUbFT8p/plugins/imageformats/libqico.so"
loaded library "/tmp/.mount_pingviUbFT8p/plugins/imageformats/libqjpeg.so"
loaded library "/tmp/.mount_pingviUbFT8p/plugins/imageformats/libqsvg.so"
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/accessiblebridge" ...
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/accessiblebridge" ...
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/accessible" ...
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/accessible" ...
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick/Controls/libqtquickcontrolsplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuickControls1Plugin",
    "debug": false,
    "uri": [
        "QtQuick.Controls"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick/Controls/libqtquickcontrolsplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick/Layouts/libqquicklayoutsplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuickLayoutsPlugin",
    "debug": false,
    "uri": [
        "QtQuick.Layouts"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick/Layouts/libqquicklayoutsplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtGraphicalEffects/libqtgraphicaleffectsplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtGraphicalEffectsPlugin",
    "debug": false,
    "uri": [
        "QtGraphicalEffects"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtGraphicalEffects/libqtgraphicaleffectsplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/Qt/labs/settings/libqmlsettingsplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QmlSettingsPlugin",
    "debug": false,
    "uri": [
        "Qt.labs.settings"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/Qt/labs/settings/libqmlsettingsplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick/Dialogs/libdialogplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuick2DialogsPlugin",
    "debug": false,
    "uri": [
        "QtQuick.Dialogs"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick/Dialogs/libdialogplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtQuick/PrivateWidgets/libwidgetsplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtQuick2PrivateWidgetsPlugin",
    "debug": false,
    "uri": [
        "QtQuick.PrivateWidgets"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtQuick/PrivateWidgets/libwidgetsplugin.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtGraphicalEffects/private/libqtgraphicaleffectsprivate.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtGraphicalEffectsPrivatePlugin",
    "debug": false,
    "uri": [
        "QtGraphicalEffects.private"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtGraphicalEffects/private/libqtgraphicaleffectsprivate.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/qml/QtCharts/libqtchartsqml2.so, metadata=
{
    "IID": "org.qt-project.Qt.QQmlExtensionInterface/1.0",
    "MetaData": {
    },
    "className": "QtChartsQml2Plugin",
    "debug": false,
    "uri": [
        "QtCharts"
    ],
    "version": 330241
}


loaded library "/tmp/.mount_pingviUbFT8p/qml/QtCharts/libqtchartsqml2.so"
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/iconengines" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/iconengines/libqsvgicon.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/iconengines/libqsvgicon.so, metadata=
{
    "IID": "org.qt-project.Qt.QIconEngineFactoryInterface",
    "MetaData": {
        "Keys": [
            "svg",
            "svgz",
            "svg.gz"
        ]
    },
    "className": "QSvgIconPlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("svg", "svgz", "svg.gz")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/iconengines" ...
ping.waterfall: Gradients: ("Thermal 5", "Thermal 6", "Thermal 7", "Monochrome", "Ocean", "Transparent", "Fishfinder", "Rainbow")
Probing UDP "192.168.2.2" 9090
ping.logger: "qml" true
ping.logger: "qml" true
ping.logger: filter "qml.debug=true\nping.waterfallGradient.debug=true\nping.waterfall.debug=true\nping.logger.debug=true\nping.util.debug=true\nping.main.debug=true\n"
ping.logger: "ping.waterfallGradient" true
ping.logger: "ping.waterfallGradient" true
ping.logger: filter "qml.debug=true\nping.waterfallGradient.debug=true\nping.waterfall.debug=true\nping.logger.debug=true\nping.util.debug=true\nping.main.debug=true\n"
ping.logger: "ping.waterfall" true
ping.logger: "ping.waterfall" true
ping.logger: filter "qml.debug=true\nping.waterfallGradient.debug=true\nping.waterfall.debug=true\nping.logger.debug=true\nping.util.debug=true\nping.main.debug=true\n"
ping.logger: "ping.logger" true
ping.logger: "ping.logger" true
ping.logger: filter "qml.debug=true\nping.waterfallGradient.debug=true\nping.waterfall.debug=true\nping.logger.debug=true\nping.util.debug=true\nping.main.debug=true\n"
ping.logger: "ping.util" true
ping.logger: "ping.util" true
ping.logger: filter "qml.debug=true\nping.waterfallGradient.debug=true\nping.waterfall.debug=true\nping.logger.debug=true\nping.util.debug=true\nping.main.debug=true\n"
ping.logger: "ping.main" true
ping.logger: "ping.main" true
ping.logger: filter "qml.debug=true\nping.waterfallGradient.debug=true\nping.waterfall.debug=true\nping.logger.debug=true\nping.util.debug=true\nping.main.debug=true\n"
Probing Serial "ttyUSB0"
Ping detected on "ttyUSB0" 115200
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/plugins/bearer" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/bearer/libqconnmanbearer.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/bearer/libqconnmanbearer.so, metadata=
{
    "IID": "org.qt-project.Qt.QBearerEngineFactoryInterface",
    "MetaData": {
        "Keys": [
            "connman"
        ]
    },
    "className": "QConnmanEnginePlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("connman")
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/bearer/libqgenericbearer.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/bearer/libqgenericbearer.so, metadata=
{
    "IID": "org.qt-project.Qt.QBearerEngineFactoryInterface",
    "MetaData": {
        "Keys": [
            "generic"
        ]
    },
    "className": "QGenericEnginePlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("generic")
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_pingviUbFT8p/plugins/bearer/libqnmbearer.so"
Found metadata in lib /tmp/.mount_pingviUbFT8p/plugins/bearer/libqnmbearer.so, metadata=
{
    "IID": "org.qt-project.Qt.QBearerEngineFactoryInterface",
    "MetaData": {
        "Keys": [
            "networkmanager"
        ]
    },
    "className": "QNetworkManagerEnginePlugin",
    "debug": false,
    "version": 330241
}


Got keys from plugin meta data ("networkmanager")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviUbFT8p/bearer" ...
loaded library "/tmp/.mount_pingviUbFT8p/plugins/bearer/libqconnmanbearer.so"
loaded library "/tmp/.mount_pingviUbFT8p/plugins/bearer/libqgenericbearer.so"
loaded library "/tmp/.mount_pingviUbFT8p/plugins/bearer/libqnmbearer.so"
loaded library "crypto"
loaded library "ssl"
qt.network.ssl: QSslSocket: cannot resolve SSLv3_client_method
qt.network.ssl: QSslSocket: cannot resolve SSLv3_server_method
```

##strace:
```
munmap(0x7f6cb9a2a000, 23544)           = 0
write(2, "Got keys from plugin meta data ("..., 47Got keys from plugin meta data ("svg", "svgz")
) = 47
write(2, "QFactoryLoader::QFactoryLoader()"..., 101QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_pingviSshrCi/imageformats" ...
) = 101
access("/tmp/.mount_pingviSshrCi/imageformats/.", F_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/libqgif.so", O_RDONLY|O_CLOEXEC) = 11
read(11, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
fstat(11, {st_mode=S_IFREG|0644, st_size=27488, ...}) = 0
mmap(NULL, 29208, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 11, 0) = 0x7f6cb9a28000
mmap(0x7f6cb9a2e000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 11, 0x5000) = 0x7f6cb9a2e000
close(11)                               = 0
mprotect(0x7f6cb9a2e000, 4096, PROT_READ) = 0
write(2, "loaded library \"/tmp/.mount_ping"..., 74loaded library "/tmp/.mount_pingviSshrCi/plugins/imageformats/libqgif.so"
) = 74
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/libqico.so", O_RDONLY|O_CLOEXEC) = 11
read(11, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
fstat(11, {st_mode=S_IFREG|0644, st_size=31744, ...}) = 0
mmap(NULL, 33464, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 11, 0) = 0x7f6cb9a1f000
mprotect(0x7f6cb9a25000, 4096, PROT_NONE) = 0
mmap(0x7f6cb9a26000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 11, 0x6000) = 0x7f6cb9a26000
close(11)                               = 0
mprotect(0x7f6cb9a26000, 4096, PROT_READ) = 0
write(2, "loaded library \"/tmp/.mount_ping"..., 74loaded library "/tmp/.mount_pingviSshrCi/plugins/imageformats/libqico.so"
) = 74
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/libqjpeg.so", O_RDONLY|O_CLOEXEC) = 11
read(11, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
fstat(11, {st_mode=S_IFREG|0644, st_size=347640, ...}) = 0
mmap(NULL, 349272, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 11, 0) = 0x7f6cb8004000
mprotect(0x7f6cb8057000, 4096, PROT_NONE) = 0
mmap(0x7f6cb8058000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 11, 0x53000) = 0x7f6cb8058000
close(11)                               = 0
mprotect(0x7f6cb8058000, 4096, PROT_READ) = 0
write(2, "loaded library \"/tmp/.mount_ping"..., 75loaded library "/tmp/.mount_pingviSshrCi/plugins/imageformats/libqjpeg.so"
) = 75
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/libqsvg.so", O_RDONLY|O_CLOEXEC) = 11
read(11, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
fstat(11, {st_mode=S_IFREG|0644, st_size=23544, ...}) = 0
mmap(NULL, 25272, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 11, 0) = 0x7f6cb9a18000
mmap(0x7f6cb9a1d000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 11, 0x4000) = 0x7f6cb9a1d000
close(11)                               = 0
openat(AT_FDCWD, "/opt/oce/lib/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/opt/opencascade/lib/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/tls/haswell/x86_64/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/tls/haswell/x86_64", 0x7ffe2f9cefc0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/tls/haswell/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/tls/haswell", 0x7ffe2f9cefc0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/tls/x86_64/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/tls/x86_64", 0x7ffe2f9cefc0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/tls/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/tls", 0x7ffe2f9cefc0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/haswell/x86_64/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/haswell/x86_64", 0x7ffe2f9cefc0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/haswell/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/haswell", 0x7ffe2f9cefc0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/x86_64/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/x86_64", 0x7ffe2f9cefc0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/tmp/.mount_pingviSshrCi/plugins/imageformats/../../lib/libQt5Svg.so.5", O_RDONLY|O_CLOEXEC) = 11
read(11, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0\0\0\0\0\0\0\0"..., 832) = 832
fstat(11, {st_mode=S_IFREG|0644, st_size=359136, ...}) = 0
mmap(NULL, 360848, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 11, 0) = 0x7f6cb341f000
mprotect(0x7f6cb3473000, 4096, PROT_NONE) = 0
mmap(0x7f6cb3474000, 16384, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 11, 0x54000) = 0x7f6cb3474000
close(11)                               = 0
mprotect(0x7f6cb3474000, 8192, PROT_READ) = 0
mprotect(0x7f6cb9a1d000, 4096, PROT_READ) = 0
write(2, "loaded library \"/tmp/.mount_ping"..., 74loaded library "/tmp/.mount_pingviSshrCi/plugins/imageformats/libqsvg.so"
) = 74
brk(0x1cc2000)                          = 0x1cc2000
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
getpid()                                = 27873
poll([{fd=5, events=POLLIN|POLLOUT}], 1, -1) = 1 ([{fd=5, revents=POLLOUT}])
writev(5, [{iov_base="\233\4\2\0\6\0 \7\233\34\2\0\1\0 \0076\30\2\0\4\0 \7", iov_len=24}, {iov_base=NULL, iov_len=0}, {iov_base="", iov_len=0}], 3) = 24
getpid()                                = 27873
```

**Comment by patrickelectric** [comment]
PS: It works with .png

**Comment by magisterRab** [comment]
I have the same problem. You nead to add 
QtSvg library
plugin iconengines/libqsvgicon.dll
plugin imageformats/libqsvg.dll

**Comment by magisterRab** [comment]
Unfortunately, I'm not familiar with the details of your project

**Comment by TheAssassin** [comment]
Okay, but thanks anyway. Perhaps @patrickelectric or @probonopd will send a PR.

**Comment by patrickelectric** [comment]
@TheAssassin the problem persist with
`-extra-plugins=imageformats/libqsvg.so,iconengines/libqsvgicon.so`

**Comment by ogoffart** [comment]
The problem is that linuxdeployqt explicitly blacklist the svg image plugin, and only installs it if the main binary links against QtSvg:

https://github.com/probonopd/linuxdeployqt/blob/e03847c81da65a563b5db10aa1a39f01ba0fed83/tools/linuxdeployqt/shared.cpp#L1221

I think this is wrong.  

In most cases, the main binary don't link against QtSvg directly. The programmer just reference ".svg" file from the application and the plugin gets loaded.

**Comment by probonopd** [comment]
Good catch @ogoffart. Which brings me to the question, is there any way for `linuxdeployqt` (short of looking at the source code) to know whether an application is using the svg plugin?

@patrickelectric can you confirm that the problem goes away if you use the `-extra-plugins=iconengines,imageformats` argument?

**Comment by ogoffart** [comment]
No, it is not possible to know automatically if an application uses such a plugin.
Most application could use svg icons. For example if they use QIcon::fromTheme is the desktop is using an icon theme.
If they have any .svg assets they are quite likely to use one of these plugins.

Most application don't use QtSvg directly, just rely for the plugin to be automatically loaded for these mime types.  
In other words, I believe you must include the plugin all the time.

**Comment by probonopd** [comment]
Thanks @ogoffart. Can you confirm that the problem goes away if you use the `-extra-plugins=iconengines,imageformats` argument?

**Comment by probonopd** [comment]
Ping @ogoffart

**Comment by ogoffart** [comment]
In my case, I made the problem disapear by manually copying the plugin in the appimage. I did not try adding the argument (and i won't because i am not working with that anymore)

In my opinion, svg should not get handled specially in the `deployPlugins` function.

**Comment by probonopd** [comment]
Which is what I am doing in https://github.com/probonopd/linuxdeployqt/commit/eba669d39b7f5d7ee67cf1e4ee4e7c457d19430e

**Comment by probonopd** [comment]
@ogoffart please retry with the latest continuous build. Does it resolve the issue?

**Comment by patrickelectric** [comment]
@probonopd 	tested and working with `extra-plugins=iconengines,imageformats`.
